src/main/java/org/plumelib/javadoc/RequireJavadoc.java

```diff
@@ -289,6 +320,158 @@ boolean shouldExclude(Path path) {
+    String methodName = md.getNameAsString();
+    if (methodName.startsWith("get")) {
+      String propertyName = propertyName(md, true);
+      if (propertyName == null) {
+        return false;
+      }
+      Statement statement = getOnlyStatement(md);
+      if (!(statement instanceof ReturnStmt)) {
+        return false;
+      }
+      Optional<Expression> oReturnExpr = ((ReturnStmt) statement).getExpression();
+      if (!oReturnExpr.isPresent()) {
+        return false;
+      }
+      Expression returnExpr = oReturnExpr.get();
+      String returnName;
+      if (returnExpr instanceof NameExpr) {
+        returnName = ((NameExpr) returnExpr).getNameAsString();
+      } else if (returnExpr instanceof FieldAccessExpr) {
+        FieldAccessExpr fa = (FieldAccessExpr) returnExpr;
+        Expression receiver = fa.getScope();
+        if (!(receiver instanceof ThisExpr)) {
+          return false;
+        }
+        returnName = fa.getNameAsString();
+      } else {
+        return false;
+      }
+      if (!returnName.equals(propertyName)) {
+        return false;
+      }
+      return true;
+    } else if (methodName.startsWith("set")) {
+      String propertyName = propertyName(md, false);
+      if (propertyName == null) {
+        return false;
+      }
+      Statement statement = getOnlyStatement(md);
+      if (!(statement instanceof ExpressionStmt)) {
+        return false;
+      }
+      Expression expr = ((ExpressionStmt) statement).getExpression();
+      if (!(expr instanceof AssignExpr)) {
+        return false;
+      }
+      AssignExpr assignExpr = (AssignExpr) expr;
+      Expression target = assignExpr.getTarget();
+      AssignExpr.Operator op = assignExpr.getOperator();
+      Expression value = assignExpr.getValue();
+      if (!(target instanceof FieldAccessExpr)) {
+        return false;
+      }
+      FieldAccessExpr fa = (FieldAccessExpr) target;
+      Expression receiver = fa.getScope();
+      if (!(receiver instanceof ThisExpr)) {
+        return false;
+      }
+      if (!fa.getNameAsString().equals(propertyName)) {
+        return false;
+      }
+      if (op != AssignExpr.Operator.ASSIGN) {
+        return false;
+      }
+      if (!(value instanceof NameExpr
+          && ((NameExpr) value).getNameAsString().equals(propertyName))) {
+        return false;
+      }
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Returns the lower-cased name of the property, if the method is a getter or setter. Otherwise
+   * returns null.
+   *
+   * @param md the method to test
+   * @param isGetter true if testing for a getter, false if testing for a setter
+   * @return the lower-cased names of the property, or null
+   */
+  @Nullable String propertyName(MethodDeclaration md, boolean isGetter) {
+    String methodName = md.getNameAsString();
+    if (!methodName.startsWith(isGetter ? "get" : "set")) {
+      return null;
+    }
+    String upperCaseProperty = methodName.substring(3);
```

Comment:
Done.  Thanks for the suggestion.

I don't think many people remember programming in Pascal -- I haven't done so since the mid-80s.  :-)

Reply:
> I don't think many people remember programming in Pascal -- I haven't done so since the mid-80s. :-)

Wrote my master's thesis code with it in the mid-90s! 😁